src/main/java/core/basesyntax/handlers/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax.handlers;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.storage.Storage;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    public void process(Fruit fruit, Integer quantity) {
+        Integer initialQuantity = Storage.storage.get(fruit);
+        if (quantity >= initialQuantity) {
```

Comment:
```suggestion
        if (quantity > initialQuantity) {
```
Потрібно використати >, бо кількість фруктів яких хочуть купити може рівна кількості, яка є. Наприклад, купити 10, коли в магазині 10, у тебе в цьому випадку буде прокинута помилка.